docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,433 @@
 # KubeStellar Architecture
 
-**NOTE**: This is only a very rudimentary start.
+KubeStellar provides multi-cluster deployment of Kubernetes objects, controlled by a simple 
+placement policy, where Kubernetes objects are expressed in their native format with no wrapping
+or bundling. The high-level architecture for KubeStellar is illustrated in Figure 1.
 
-## Higher Level
+<figure>
+  <img src="./high-level-architecture.png"  alt="High Level Architecture">
+  <figcaption align="center">Figure 1 - High Level Architecture </figcaption>
+</figure>
 
-At the higher level there are three modules.
+KubeStellar relies on the concept of *spaces*.  
+A Space is an abstraction to represent an API service that 
+behaves like a Kubernetes kube-apiserver (including the persistent storage behind it) 
+and the subset of controllers in the kube-controller-manager that are concerned with 
+API machinery generalities (not management of containerized workloads). 
+A KubeFlex `ControlPlane` is an example. A regular Kubernetes cluster is another example.
+Users can use spaces to perform these tasks:
 
-1. KubeStellar
-2. [the status addon](https://github.ibm.com/dettori/status-addon)
-3. KubeFlex
+1. Create *Workload Definition Spaces* (WDSs) to store the definitions of their workloads.
+A Kubernetes workload is an application that runs on Kubernetes. A workload can be made by a 
+single Kubernetes object or several objects that work together.
+2. Create *Inventory and Transport Spaces* (ITSs) to manage the inventory of clusters and 
+the transport of workloads.
+3. Register and label Workload Execution Clusters (WECs) with the Inventory and 
+Transport Space, to keep track of the available clusters and their characteristics.
+4. Define Placement Policies to specify *what" objects and *where* should be 
+deployed on the WECs.
+5. Submit objects in the native Kubernetes format to the WDSs, 
+and let the Placement Policies govern which WECs should receive them.
+6. Check the status of submitted objects from the WDS.
 
-The last two are independent and KubeStellar depends on them.
+In KubeStellar, users can assume a variety of roles and responsibilities. 
+These roles could range from system administrators and application owners 
+to CISOs and DevOps Engineers. However, for the purpose of this document, 
+we will not differentiate between these roles. Instead, we will use the 
+term ‘user’ to collectively refer to any individual who encompasses all these roles.
```

Comment:
"encompasses all these roles" is restrictive --- it excludes individuals who play a strict subset of those roles. I think you want to say something like "any individual in any of those roles". But that would still imply that we think that we have enumerated a definitive set of roles. I think the best approach is something like the following.

> Instead we will use the term 'user' broadly, without attempting to make distinctions among roles.

Reply:
done